lib/SILOptimizer/PassManager/PassPipeline.cpp

```diff
@@ -477,6 +477,9 @@ SILPassPipelinePlan::getPerformancePassPipeline(const SILOptions &Options) {
   // stdlib.
   addPerfEarlyModulePassPipeline(P);
 
+  // Cleanup after SILGen: remove trivial copies to temporaries.
+  P.addTempRValueOpt();
```

Comment:
Move `P.addTempRValueOpt();` into `addPerfEarlyModulePassPipeline` or `addHighLevelEarlyLoopOptPipeline `? We don't seem to add any passes explicitly inside getPerformancePassPipeline and usually do it inside helper functions.